src/components/location/location.service.ts

```diff
@@ -381,7 +385,9 @@ export class LocationService {
     }
 
     if (!result.canReadRegion) {
-      throw new Error('User does not have permission to read this region');
+      throw new UnauthorizedException(
```

Comment:
```suggestion
      throw new ForbiddenException(
```
I know the naming is weird 😞